src/views/AddItem.jsx

```diff
@@ -1,39 +1,80 @@
 import { useEffect, useState } from 'react';
 import { addItem } from '../api/firebase';
 
-export function AddItem({ listId }) {
+export function AddItem({ data, listId }) {
 	const [timeframe, setTimeframe] = useState('7');
 	const [itemName, setItemName] = useState('');
-	const [message, setMessage] = useState(null);
+	const [messages, setMessages] = useState([]);
 
 	useEffect(() => {
 		const timer = setTimeout(() => {
-			setMessage(null);
+			setMessages([]);
 		}, 3000);
 
 		return () => clearTimeout(timer);
-	}, [message]);
+	}, [messages]);
 
 	const onTimeChange = (e) => setTimeframe(e.target.value);
 	const onItemChange = (e) => setItemName(e.target.value);
+
+	// function to validate itemName input that returns error messages if any - called onFormSubmit
+	function validateItemInput(data, trimmedItemName) {
+		// 1- checks for empty inputs
+
+		if (!trimmedItemName) {
+			return ['Please enter an item name.'];
+		}
+
+		const errArray = [];
+
+		// 2- checks for invalid characters
+
+		if (trimmedItemName.search(/[^a-z0-9'&-\s]/i) !== -1) {
+			errArray.push(
+				`Please enter an item name that is alphanumeric or includes ', -, and &.`,
```

Comment:
I think rephrasing this to something like `Invalid item name entered. Please only use alphanumeric, ', -, or & characters in your item name` might make it more clear? It's mainly the "or" but it could just be me.